webserver.py

```diff
@@ -60,14 +60,23 @@ def do_GET(self):
             logger.warn("Connected to from " + str(self.client_address) + " looking at " + str(inst))
 
             with _scraped_data_lock:
-                if inst not in _scraped_data.keys():
-                    raise ValueError(str(inst) + " not known")
-                if _scraped_data[inst] == "":
-                    raise ValueError("Instrument has become unavailable")
-                try:
-                    ans = "%s(%s)" % (callback, json.dumps(_scraped_data[inst]))
-                except Exception as err:
-                    raise ValueError("Unable to convert instrument data to JSON: %s" % err.message)
+                active = {}
+                if inst == "ALL":
+                    for key in _scraped_data:
+                        if _scraped_data[key] != "":
+                            active[key] = True
+                        else:
+                            active[key] = False
+                    ans = "%s(%s)" % (callback, json.dumps(active))
+                else:
+                    if inst not in _scraped_data.keys():
+                        raise ValueError(str(inst) + " not known")
+                    if _scraped_data[inst] == "":
+                        raise ValueError("Instrument has become unavailable")
+                    try:
+                        ans = "%s(%s)" % (callback, json.dumps(_scraped_data[inst]))
+                    except Exception as err:
+                        raise ValueError("Unable to convert instrument data to JSON: %s" % err.message)
```

Comment:
These two blocks do very different things.

- For all instruments, it returns whether the instrument is active
- For specific instruments, it returns all of the data about that instrument

That difference in behaviour should be highlighted more clearly within the code. The two conditional blocks should be abstracted into functions.